test/jasmine/tests/colorscale_test.js

```diff
@@ -368,6 +369,117 @@ describe('Test colorscale:', function() {
         });
     });
 
+    describe('handleDefaults (coloraxis version)', function() {
+        it('should not coerced colorscale/colorbar attributes when referencing a shared color axis', function() {
+            var gd = _supply([
+                {type: 'heatmap', z: [[0]]},
+                {type: 'heatmap', z: [[2]], coloraxis: 'coloraxis'},
+                {type: 'heatmap', z: [[2]], coloraxis: 'coloraxis'},
+            ]);
+
+            var fullData = gd._fullData;
+            var fullLayout = gd._fullLayout;
+
+            var zAttrs = ['zauto', 'colorscale', 'reversescale'];
+            zAttrs.forEach(function(attr) {
+                expect(fullData[0][attr]).not.toBe(undefined, 'trace 0 ' + attr);
+                expect(fullData[1][attr]).toBe(undefined, 'trace 1 ' + attr);
+                expect(fullData[2][attr]).toBe(undefined, 'trace 2 ' + attr);
+            });
+
+            var cAttrs = ['cauto', 'colorscale', 'reversescale'];
+            cAttrs.forEach(function(attr) {
+                expect(fullLayout.coloraxis[attr]).not.toBe(undefined, 'coloraxis ' + attr);
```

Comment:
Please note that `layout.coloraxis?` use `cmin`, `cmax` and `cauto` no matter which trace types make reference to them.  In other words, setting `zmin` or `zmax` in a coloraxis won't do anything.